part2/diagonalization.py

```
@@ -1,37 +1,414 @@
+- Phát hiện trị riêng phức (trong trường hợp hỗ trợ)
+- Sai số tái tạo vượt ngưỡng cho phép
+
+Phù hợp cho ma trận kích thước nhỏ–trung bình với phổ thực.
+"""
+
+try:
+    from part1.inverse import inverse
+except ImportError:
+    from part1.inverse import inverse
+
+from __future__ import annotations
```

Ruff (0.15.9)

[error] `from __future__` imports must occur at the beginning of the file

(F404)